### DISCUSSION TITLE
tiff conversion to pdf is occasionally inverting the image to mirror

### DISCUSSION THREAD

**Discussion by D01063197** [ImageMagick/ImageMagick#2682] [Help]
I'm using ImageMagick to merge multiple .tif files and convert them to one .pdf file. Occasionally I'm seeing a mirror image produced during the merge/conversion process. I see the following error below, but the files are actually created, but in mirror image format. Is there any way to either prevent this from happening or convert again if the mirror image happens?

Thanks!

Andy

"Invoke Power Shell: PowerShell command errors:
magick.exe: Unknown field with tag 32934 (0x80a6) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1044.
magick.exe: Deprecated and troublesome old-style JPEG compression mode, please convert to new-style JPEG compression and notify vendor of writing software. `OJPEGSetupDecode' @ warning/tiff.c/TIFFWarnings/1044."

**Comment by snibgo** [comment]
What version of IM?

To help you, we need to reproduce the problem. Please show your command, and an input tif file that creates the problem, and the output pdf.

> I see the following error below, ...

That is a warning, not an error.

**Comment by D01063197** [comment]
Command Line and Version below. Files attached.

C:\temp\Andy>C:\tools\ImageMagick\magick.exe *.tif OfficialTranscript_1_470334.pdf
magick.exe: Unknown field with tag 32934 (0x80a6) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1044.
magick.exe: Deprecated and troublesome old-style JPEG compression mode, please convert to new-style JPEG compression and notify vendor of writing software. `OJPEGSetupDecode' @ warning/tiff.c/TIFFWarnings/1044.

C:\temp\Andy>dir
Volume in drive C has no label.
Volume Serial Number is 162A-5F2D

Directory of C:\temp\Andy

10/06/2020  09:16 AM    <DIR>          .
10/06/2020  09:16 AM    <DIR>          ..
10/06/2020  09:16 AM         1,334,869 OfficialTranscript_1_470334.pdf
10/06/2020  07:49 AM           445,004 OfficialTranscript_1_Page1_470334_.tif
10/06/2020  07:49 AM           607,992 OfficialTranscript_1_Page2_470334_.tif
               3 File(s)      2,387,865 bytes
               2 Dir(s)  81,138,511,872 bytes free

Version:

C:\SOCOE\Post Licensure\Output\OfficialTranscripts\D10020044>C:\tools\ImageMagick\magick.exe -version
Version: ImageMagick 7.0.10-24 Q16 x64 2020-07-18 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 192628806
Features: Cipher DPC OpenCL
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib

**Comment by snibgo** [comment]
To reproduce the problem, I need to download your files. The actual files, not just a directory listing.

**Reply by D01063197** [reply]
[Uploading OfficialTranscript_1_Page1_470334_.zip…]()

**Comment by D01063197** [comment]
I attached them to my last reply in email. I guess it doesn't accept attachments. Had to zip the files because it wouldn't take a .tiff file.

**Comment by snibgo** [comment]
The URL in your link is simply to this page, not to your zip file.

**Comment by D01063197** [comment]
How can I get the files to you?

**Comment by snibgo** [comment]
Upload the zip file to somewhere on the internet, and paste the URL here.

**Reply by D01063197** [reply]
https://filebin.net/gxw2g6oo4jx691qp/OfficialTranscript_1_Page1_470334_.zip?t=qlx7gbjm

**Comment by snibgo** [comment]
Your tif files both have metadata orientation "right-top", aka "Rotate 90 CW". But your first image OfficialTranscript_1_Page1_470334_.tif also needs to be flipped. So the metadata is wrong. You could correct it manually with exiftool.

Automatically correcting the metadata would not be simple. Perhaps attempt to OCR the image (eg with tesseract), then OCR it after flipping, and measure which reads the most text.

**Comment by D01063197** [comment]
Couple more questions. When I open the .tif file “OfficialTranscript_1_Page1_470334_.tif”, it is not inverted. Once I run ImageMagick on the two .tif files, page one comes out inverted. What in the metadata is telling the tool to invert the image? Is there a parameter to the tool to not invert the image during the pdf conversion?

**Comment by snibgo** [comment]
> When I open the .tif file “OfficialTranscript_1_Page1_470334_.tif”, it is not inverted. 

When you open the file with what software? Does that software respect or ignore the orientation metadata?

If I open it with Gimp v2.10.0, I am offered the choice of "Original" (which ignores the orientation metadata) or "Rotated" (which actions the metadata, like IM's "-auto-orient"). In both of these, the text reads backwards.

> What in the metadata is telling the tool to invert the image?

The orientation flag. Use exiftool of "magick identify -verbose" to see this.

> Is there a parameter to the tool to not invert the image during the pdf conversion?

IM ignores the orientation metadata, unless you use "-auto-orient".

IM is not inverting the image. It is already inverted, in the tif file.

**Comment by fmw42** [comment]
Inverted in geometry or inverted in color?

**Comment by snibgo** [comment]
In geometry. The text is most easily read with the help of a mirror.
